yew-macro/src/html_tree/html_tag/mod.rs

```diff
@@ -137,95 +133,192 @@ impl ToTokens for HtmlTag {
+            Some(ClassesForm::Single(classes)) => match classes.try_into_lit() {
+                Some(lit) => {
+                    if lit.value().is_empty() {
+                        None
+                    } else {
+                        let sr = lit.stringify();
+                        Some(quote! {
+                            #vtag.__macro_push_attribute(::yew::virtual_dom::PositionalAttr::new("class", #sr));
+                        })
+                    }
+                }
+                None => {
+                    let sr = stringify::stringify_dynamic(quote! { __yew_classes });
+                    Some(quote! {
+                        let __yew_classes = ::std::convert::Into::<::yew::virtual_dom::Classes>::into(#classes);
+                        if !__yew_classes.is_empty() {
+                            #vtag.__macro_push_attribute(::yew::virtual_dom::PositionalAttr::new("class", #sr));
+                        };
+                    })
+                }
             },
             None => None,
         };
-        let set_classes_it = set_classes.iter();
 
-        let set_node_ref = node_ref.iter().map(|node_ref| {
-            quote! {
-                #vtag.node_ref = #node_ref;
-            }
-        });
-        let set_key = key.iter().map(|key| {
-            quote! {
-                #vtag.key = Some(::std::convert::Into::<::yew::virtual_dom::Key>::into(#key));
-            }
-        });
-        let listeners: Vec<_> = listeners
-            .iter()
-            .map(|listener| {
-                let name = &listener.label.name;
-                let callback = &listener.value;
-
-                quote_spanned! {name.span()=> ::yew::html::#name::Wrapper::new(
-                    <::yew::virtual_dom::VTag as ::yew::virtual_dom::Transformer<_, _>>
-                        ::transform(#callback),
-                )}
+        let add_listeners = if listeners.is_empty() {
+            None
+        } else if listeners.iter().any(|attr| attr.question_mark.is_some()) {
+            let add_listeners = listeners
+                .iter()
+                .map(
+                    |TagAttribute {
+                         label,
+                         question_mark,
+                         value,
+                     }| {
+                        let name = &label.name;
+
+                        if question_mark.is_some() {
+                            let ident = Ident::new("__yew_listener", name.span());
+                            let listener = generate_listener_code(name, &ident);
+                            quote_spanned! {value.span()=>
+                                let #ident = ::std::option::Option::map(#value, |#ident| {
+                                    #listener
+                                });
+                                if let ::std::option::Option::Some(#ident) = #ident {
+                                    #vtag.add_listener(#ident);
+                                };
```

Comment:
Yes, but it generates a worse error message. Instead of saying "expected an Option, found something else" it will say "expected <type of value>, found <fully qualified Option type>"

Reply:
Ah got it, sounds good!